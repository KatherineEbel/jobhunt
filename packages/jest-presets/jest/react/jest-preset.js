module.exports = {
  roots: ['<rootDir>/src'],
  globals: {
    jest: true,
  },
  testEnvironment: 'jsdom',
  transform: {
    '^.+\\.tsx?$': 'ts-jest',
  },
  moduleDirectories: [
    "node_modules", "src",
  ],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
  modulePathIgnorePatterns: [
    '<rootDir>/test/__fixtures__',
    '<rootDir>/node_modules',
    '<rootDir>/dist',
  ],
  preset: 'ts-jest',
  setupFilesAfterEnv: ['<rootDir>/jest-setup.ts']
}
