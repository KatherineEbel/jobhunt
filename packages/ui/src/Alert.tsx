import React, {useEffect, useState} from 'react'
import styled from 'styled-components'
import { Alert as JHAlert } from 'lib'


const StyledAlert = styled.p<{show: boolean}>`
  border-radius: var(--borderRadius);
  overflow: hidden;
  padding: 0.6rem 1.5rem;
  transition: transform 300ms ease-in-out;
  transform: scaleY(${({show}) => show ? 1 : 0});
`

const AlertDanger = styled(StyledAlert)`
  background: var(--red-light);
  color: var(--red-dark);
`

const AlertSuccess = styled(StyledAlert)`
  background: var(--green-light);
  color: var(--green-dark);
`

export const Alert = ({alerts}: {alerts: JHAlert[]}) => {
  const [alert, setAlert] = useState<JHAlert | null>()
  const [show, setShow] = useState(false)

  useEffect(() => {
    if (alerts.length > 0) {
      setAlert(alerts[0])
      setShow(true)
      setTimeout(() => {
        setShow(false)
      }, 5000)
    }
  }, [alerts])

  if (!alert) return null

  const Component = alert.type === 'success' ? AlertSuccess : AlertDanger

  return <Component show={show} data-test-id='alert' className='alert'>{alert.message}</Component>
}
