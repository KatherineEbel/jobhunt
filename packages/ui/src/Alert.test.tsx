import "@testing-library/jest-dom";
import {render, screen} from '@testing-library/react'
import React from 'react'
import {Alert} from './Alert'

describe('Alert', function () {
  test('render', () => {
    render(<Alert message='Test message' type='success'/>)
    const alert = screen.getByText('Test message')
    expect(alert).toBeInTheDocument()
  })
})