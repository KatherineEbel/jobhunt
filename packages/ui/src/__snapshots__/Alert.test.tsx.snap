// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert render 1`] = `<DocumentFragment />`;
